decimal.go

```diff
@@ -265,11 +269,12 @@ func (d Decimal) Div(d2 Decimal) Decimal {
 	return ret
 }
 
-// Cmp compares x and y and returns -1, 0 or 1
+// Cmp compares d and d2 and returns -1, 0 or 1.
+// The smallest scale of d and d2 are used in comparison.
```

Comment:
This comment is also confusing, just say it compares the numbers represented by the decimals, regardless of what their internal representation is.